### DISCUSSION TITLE
Results Webpart Still Removing Duplicates

### DISCUSSION THREAD

**Discussion by Jayb127** [microsoft-search/pnp-modern-search#2260] [Q&A]
Sorry if this isn't the right place to ask this, but I'm wondering if anyone else is experiencing issues with the removal of "duplicate" results in the Search Results webpart. 

I found this discussion: https://github.com/microsoft-search/pnp-modern-search/discussions/2088

Which appears to have prompted a solution. However, changing this setting is not resolving my issue, and duplicates are still being removed from my results, even though they're not actually duplicates. I've built a search configuration with a pretty narrow scope and limited test data, but I'm already seeing instances of duplicate removal. The results webpart indicates that there are 10 results, but only 8 appear in the webpart. The content in question is a custom content type inheriting inheriting from the OOTB link webpart, so there aren't a ton of options for creating wide variations between the list items like there might be in a document library. 

Here are some other observations that may help:
- Occasionally, and seemingly randomly, 1 or more of the missing results will appear in the results webpart. This happens less than 25% of the time. Refreshing the page usually causes the result to disappear.
- Selecting filters will sometimes (again, about 25% of the time) make the missing result(s) appear, but the result count at the top of the results webpart will always count that missing result. For example, after selecting an applicable filter, the count will say "4 results," but only 3 results will appear.
- Performing a keyword search will force the missing result to appear in the results every time.

Are there any other options or workarounds to force these missing results to appear? Is there anything anyone can recommend that would allow me to verify that the problem here is SharePoint's interpretation of these items as duplicates as opposed to some other problem?

Thank you.

**Comment by FranckyC** [comment]
Hi @Jayb127, are you using SharePoint or Microsoft Search? If SharePoint search, you can use Search Query Tool with the same exact query and see if it the same results. I suspect this is the OOTB API behavior. The way the search engine determines duplicates is up to Microsoft and we don't have any control over that unfortunately.

**Reply by Jayb127** [reply]
Hi @FranckyC 
Thank you for your response. 
To clarify, this is a SharePoint Search. When you refer to the Search Query Tool, can you be more specific? Are you referring to a subset of the configuration options in the Modern Results webpart Property Pane? Right now, I have the results webpart pointing at a site collection result source.

**Reply by FranckyC** [reply]
By Search Query Tool I mean this: https://github.com/pnp/PnP-Tools/tree/master/Solutions/SharePoint.Search.QueryTool

**Reply by Jayb127** [reply]
I will give this a shot and report back. Thank you for the suggestion.
